pkg/steps/promote.go

```diff
@@ -102,9 +102,25 @@ func (s *promotionStep) Run(ctx context.Context, dry bool) error {
 		if valid == nil {
 			continue
 		}
+
+		name := fmt.Sprintf("%s%s", s.config.NamePrefix, dst)
+
+		_, err := s.dstClient.ImageStreams(s.config.Namespace).Get(name, meta.GetOptions{})
```

Comment:
You can just do a blind create and ignore already exists.  That would be simpler and easier to reason about.

Reply:
And ignore forbidden (because you might have update permission but not create).

Reply:
ACK